src/database/DataTypes/StocktakeItem.js

```diff
@@ -55,48 +73,75 @@ export class StocktakeItem extends Realm.Object {
    * @return {Boolean} Whether the counted quantity is below the minimum for this item
    */
   get isReducedBelowMinimum() {
-    return this.countedTotalQuantity !== undefined &&
-           this.countedTotalQuantity !== null &&
-           this.countedTotalQuantity < this.minimumTotalQuantity;
-  }
-
-  addBatch(stocktakeBatch) {
-    this.batches.push(stocktakeBatch);
-    if (!this.countedTotalQuantity) this.countedTotalQuantity = stocktakeBatch.countedTotalQuantity;
-    else this.countedTotalQuantity += stocktakeBatch.countedTotalQuantity;
+    const countedTotalQuantity = this.countedTotalQuantity;
+    return countedTotalQuantity !== undefined &&
+           countedTotalQuantity !== null &&
+           countedTotalQuantity < this.minimumTotalQuantity;
   }
 
   /**
-   * Applies the adjustments to batches in the given transaction item to the batches
-   * in this stocktake item
-   * @param  {Realm}   database        App wide database
-   * @param  {object}  transactionItem The transaction item
-   * @return {none}
+   * Function is used when adjusting inventory for the StocktakeItem
+   * rather then individual StocktakeBatches. Logic:
+   * Increasing -> From latest expiry to earliest expiry StocktakeBatch, increase
+   * to the maximum of snapshot quantity, after that if still have quantity
+   * to increase, apply it to earlies expiry batch.
+   * Reducing -> From earliest expiry to latest expiry StocktakeBatch, decrease
+   * quantity by maximum === current counted quantity
```

Comment:
From what I understood here: "decrease quantity by maximum === current counted quantity" means that will not decrease below the current counted quantity for the batch? What is the current counted quantity? That's probably what got me here.

Reply:
I think latest/earliest can get mixed up in what they mean. I think it's because 'latest' often refers to dates in the past. Earliest/soonest/nearest and latest/furthest ? Not sure, I could be off base here.

Reply:
What the argument for doing additions from latest to earliest?

Reply:
If StocktakeItem quantity is changed to say zero, then increase (but below snapshot quantity), we should increase the last expiry batch first to follow FEFO rule.